_docs/migration_guides/upgrading_to_terragrunt_0.12.x.md

```diff
@@ -0,0 +1,174 @@
+# Upgrading to Terragrunt 0.12.x
```

Comment:
Yea, I think that there is a lot of content that can be broken out of the README. I think the best solution is to create a website to host the Terragrunt documentation (terragrunt.io) which follows the model of the Terraform documentation on terraform.io.

I am writing a PR now to explain this idea in more detail. I will link here once its up.